library.py

```diff
@@ -197,5 +202,13 @@ def edit_profile(uname):
             return render_template('profile.html', bio=new_bio)
 
 
+@app.route("/contact/<uname>", methods=['GET', 'POST'])
+def contact(uname):
+    if request.method == "GET":  # regular get, present the form to user to edit.
+        return render_template('contact.html', staff=uname)
+
+    elif request.method == "POST":  # form was submitted, update database
+        return redirect('/profile/' + uname)
```

Comment:
Yeah... I tried that originally but couldn't figure out the proper structure for how to add the uname reference.... it's actually `return redirect(url_for('profile', uname=uname))`.  I figured it out based on the syntax you used for the href link.  Thanks!